examples/Locomotion/Basic_Thumbsticks/main.lua

```diff
@@ -0,0 +1,119 @@
+}
+
+function motion.smooth(dt)
+  if lovr.headset.isTracked('right') then
+    local x, y = lovr.headset.getAxis('right', 'thumbstick')
+    if math.abs(x) > motion.thumbstickDeadzone then
+      motion.pose:rotate(-x * motion.turningSpeed * dt, 0, 1, 0)
+    end
+  end
+  if lovr.headset.isTracked('left') then
+    local x, y = lovr.headset.getAxis('left', 'thumbstick')
+    local direction = quat(lovr.headset.getOrientation(motion.directionFrom)):direction()
+    if not motion.flying then
+      direction.y = 0
+    end
+    if math.abs(x) > motion.thumbstickDeadzone then
+      local strafeVector = quat(-math.pi / 2, 0,1,0):mul(vec3(direction))
+      motion.pose:translate(strafeVector * x * motion.walkingSpeed * dt)
+    end
+    if math.abs(y) > motion.thumbstickDeadzone then
+      motion.pose:translate(direction * y * motion.walkingSpeed * dt)
+    end
+  end
+end
+
+function motion.snap(dt)
+  if lovr.headset.isTracked('right') then
+    local x, y = lovr.headset.getAxis('right', 'thumbstick')
+    if math.abs(x) > motion.thumbstickDeadzone and motion.thumbstickCooldown < 0 then
+      local direction = -x / math.abs(x)
+      motion.pose:rotate(motion.snapTurnAngle * direction, 0, 1, 0)
+      motion.thumbstickCooldown = motion.thumbstickCooldownTime
+    end
+  end
+  if lovr.headset.isTracked('left') then
+    local x, y = lovr.headset.getAxis('left', 'thumbstick')
+    if math.abs(y) > motion.thumbstickDeadzone and motion.thumbstickCooldown < 0 then
+      local direction = quat(lovr.headset.getOrientation('head')):direction()
+      if not motion.flying then
+        direction.y = 0
+      end
+      local moveVector = direction * motion.dashDistance * y / math.abs(y)
+      motion.pose:translate(moveVector)
+      motion.thumbstickCooldown = motion.thumbstickCooldownTime
+    end
+  end
+  motion.thumbstickCooldown = motion.thumbstickCooldown - dt
+end
+
+function lovr.update(dt)
+  motion.directionFrom = lovr.headset.isDown('left', 'trigger') and 'left' or 'head'
+  if lovr.headset.isDown('left', 'grip') then
+    motion.flying = true
+  elseif lovr.headset.wasReleased('left', 'grip') then
+    motion.flying = false
+    local landingV = vec3(motion.pose)
+    landingV.x = 0
+    landingV.z = 0
+    landingV.y = -landingV.y
+    motion.pose:translate(landingV)
+  end
+  if lovr.headset.isDown('right', 'grip') then
+    motion.snap(dt)
+  else
+    motion.smooth(dt)
+  end
+end
+
+function lovr.draw()
+  lovr.graphics.setBackgroundColor(0.1, 0.1, 0.1)
+  lovr.graphics.transform(mat4(motion.pose):invert())
+  -- Render hands
+  lovr.graphics.setColor(1,1,1)
+  local radius = 0.05
+  for _, hand in ipairs(lovr.headset.getHands()) do
+    -- Whenever pose of hand or head is used, need to account for VR movement
+    local poseRW = mat4(lovr.headset.getPose(hand))
+    local poseVR = mat4(motion.pose):mul(poseRW)
+    poseVR:scale(radius)
+    lovr.graphics.sphere(poseVR)
+  end
+  -- A pattern for spatial orientation
```

Comment:
For some reason it took me a while to figure out that this part was drawing a test scene with cylinders, from the comment here I thought it was drawing some sort of compass or debug indicator

Reply:
I'll clarify this and include some comments explaining math.

A compass in hand is actually a great idea for any locomotion example. Maybe I can add one later.